src/drive/web/modules/search/components/helpers.js

```diff
@@ -33,13 +33,18 @@ export const makeNormalizedFile = (client, folders, file) => {
   const dirId = isDir ? file._id : file.dir_id
   const urlToFolder = `${window.location.origin}/#/folder/${dirId}`
 
-  let path, url, onSelect
+  let path, url, onSelect, parentUrl
   if (isDir) {
     path = file.path
     url = urlToFolder
+    parentUrl = urlToFolder
   } else {
     const parentDir = folders.find(folder => folder._id === file.dir_id)
     path = parentDir && parentDir.path ? parentDir.path : ''
+    parentUrl =
+      parentDir && parentDir._id
+        ? `${window.location.origin}/#/folder/${parentDir._id}`
```

Comment:
@Crash-- Can we use window.location in front application ? I do not remember if we should always use URL from client or if we can use this.

Reply:
We should avoid using window.location for a lot of stuff. 

If you need to create a link to an app, you should use the `AppLinker` component from cozy-ui. 

If you need to get the `location` for an app, you should rely on cozyclient and generateWebLink for the slug.  

There is one case that it ATM difficult to handle: if you need to make a redirection without a user interaction. 

In that case, you need to test if you are in the flagship app or not and call the right method by yourself. ex: https://github.com/cozy/cozy-store/blob/6b9549a6aaac8cc388ad015bcb0034768e384230/src/ducks/apps/components/ApplicationPage/Header.jsx#L40-L56

(cc @Ldoppea)